### PR TITLE
create img_split_dota.py file for dota dataset

### PR DESCRIPTION
**Pull request by ziippy**
I want to generate annotation .txt for splitted image.
but, img_split.py create just .pkl files.

So, I create new file img_split_dota.py file.
It can make the annotation .txt file for spllitted image. 

Please review this PR.

### CODE REVIEW

File: tools/img_split_dota.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
+import BboxToolkit as bt
```
Comment (jbwang1997) [inline tools/img_split_dota.py:1]: If possible, I suggest merging this function directly into img_split.py

File: tools/img_split_dota.py [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+
+    #argument for saving
+    parser.add_argument('--filter_empty', action='store_true',
+                        help='filter out empty patches, speed up trining splitting')
+    parser.add_argument('--save_dir', type=str, default='.',
+                        help='to save pkl and splitted images')
+    parser.add_argument('--save_ext', type=str, default='.png',
+                        help='the extension of saving images')
+    parser.add_argument('--generate_txt', action='store_true', default=True,
+                        help='to generate txt for splitted images')
```
Comment (jbwang1997) [inline tools/img_split_dota.py]: It's better to not set default value for the `store_true` argument

File: tools/img_split_dota.py [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+            for i in range(len(patch_info['ann']['labels'])):
+                box = np.array2string(patch_info['ann']['bboxes'][i]).strip("[""]")
+                box += ' '
+                box += CLASSES[patch_info['ann']['labels'][i]]
+                box += ' '
+                box += str(patch_info['ann']['diffs'][i])
+                box += '\n'
+                #print(box)
+                txtfile.write(box)
+            txtfile.close()
```
Comment (jbwang1997) [inline tools/img_split_dota.py]: It's better to put txt files into a new folder. You can create a new `save_txt` folder by `save_anns = osp.join(args.save_dir, 'txt')`

File: tools/img_split_dota.py (hunk 1 above)
Comment (ziippy) [reply tools/img_split_dota.py:1]: but, It depends on DOTA CLASSES constant.
I think BboxToolkit useful many type of datasets that using Bbox annotation.
so.. I create new file new one.

File: tools/img_split_dota.py (hunk 2 above)
Comment (ziippy) [reply tools/img_split_dota.py]: ok

File: tools/img_split_dota.py (hunk 3 above)
Comment (ziippy) [reply tools/img_split_dota.py]: ok

File: tools/img_split_dota.py (hunk 1 above)
Comment (jbwang1997) [reply tools/img_split_dota.py:1]: The `classes` will be output from `load_func`. You can pass it into `single_split`.

By the way. Creating a new file is also good to understand. I accept both modifications.

It's very appreciated of you to contribute to this repo.

Comment (ziippy) [reply tools/img_split_dota.py:1]: oh! I miss the function single_split.
nextime I will try to merge directly. 

Thanks.

**Pull request merged**